src/gpio.rs

```diff
@@ -314,6 +335,27 @@ macro_rules! gpio {
                         unsafe { (*$GPIOX::ptr()).bsrr.write(|w| w.bits(1 << (16 + $i))) }
                     }
                 }
+
+                impl<MODE> OutputPin for $PXi<Alternate<MODE>> {
```

Comment:
Should we make it `UnsafeOutputPin` with all methods marked `unsafe`?
Impls like this are necessary to implement workarounds like 2.14.1 in this errata sheet: http://www.st.com/content/ccc/resource/technical/document/errata_sheet/f5/50/c9/46/56/db/4a/f6/CD00197763.pdf/files/CD00197763.pdf/jcr:content/translations/en.CD00197763.pdf.
Also it make it much more convenient to initlialize pins to proper value on startup.

Reply:
Could you elaborate on how marking the method as unsafe helps in working around the errata?

Reply:
I don't think that Alternate-mode pins should allow regular I/O in safe code, but it's needed for various workarounds. Hence I propose to make such impls unsafe.

Reply:
Oh, I hadn't notice this were the Alternate pins. I agree that this should not be public *and* safe. But, since this is working around a errata, would be possible to keep this private? I'm not 100% sure it's a good idea to expose this to users, regardless of safety.